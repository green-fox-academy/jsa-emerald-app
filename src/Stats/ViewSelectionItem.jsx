import React from 'react';
import { ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';

const ViewSelectionItem = ({ title, pressHandler }) => (
  <ListItem
    id="btn-stats-index-month"
    title={title}
    topDivider
    bottomDivider
    onPress={() => pressHandler(title.toLowerCase())}
  />
);

ViewSelectionItem.propTypes = {
  title: PropTypes.string.isRequired,
  pressHandler: PropTypes.func.isRequired,
};

export default ViewSelectionItem;
