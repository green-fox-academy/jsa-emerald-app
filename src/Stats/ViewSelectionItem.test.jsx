import React from 'react';
import { shallow } from 'enzyme';
import ViewSelectionItem from './ViewSelectionItem';

describe('<ViewSelectionItem />', () => {
  it('should render month item', () => {
    const btnClick = jest.fn();
    const wrapper = shallow(<ViewSelectionItem title="Month" pressHandler={btnClick} />);
    expect(wrapper.toString()).toMatchSnapshot();
    wrapper.simulate('press');
    expect(btnClick).toHaveBeenCalled();
  });

  it('should render year item', () => {
    const btnClick = jest.fn();
    const wrapper = shallow(<ViewSelectionItem title="Year" pressHandler={btnClick} />);
    expect(wrapper.toString()).toMatchSnapshot();
    wrapper.simulate('press');
    expect(btnClick).toHaveBeenCalled();
  });
});
