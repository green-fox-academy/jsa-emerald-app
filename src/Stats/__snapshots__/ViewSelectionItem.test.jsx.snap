// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewSelectionItem /> should render month item 1`] = `"[object Object]"`;

exports[`<ViewSelectionItem /> should render year item 1`] = `"[object Object]"`;
